src/Http/Controllers/Corporation/CorporationsController.php

```diff
@@ -47,13 +47,14 @@ public function getCorporations()
 
     /**
      * @return mixed
+     * @throws \Exception
      */
     public function getCorporationsData()
     {
 
         $corporations = $this->getAllCorporationsWithAffiliationsAndFilters();
```

Comment:
You need to pass false as parameter of this function. Otherwise every things related to paginate, order or filter will be processed on the result of a full SQL query. Which is resulting in an heavy loading time.

Sending false to it will move average transaction time from 7 seconds to 2 seconds. Must make some tests at ACL levels in order to be sure making search will not remove applied filters.

Reply:
Also, I get exceptions while attempting to sort action column or searching something.
Both are probably related.

```
DataTables warning: table id=corporations-table - Exception Message:

SQLSTATE[42S22]: Column not found: 1054 Unknown column 'actions' in 'order clause' (SQL: select * from `corporation_infos` order by `actions` asc limit 10 offset 0)
```

```
DataTables warning: table id=corporations-table - Exception Message:

SQLSTATE[42S22]: Column not found: 1054 Unknown column 'corporation_infos.actions' in 'where clause' (SQL: select count(*) as aggregate from (select '1' as `row_count` from `corporation_infos` where (LOWER(`corporation_infos`.`name`) LIKE %a% or LOWER(`corporation_infos`.`ceo_id`) LIKE %a% or LOWER(`corporation_infos`.`alliance_id`) LIKE %a% or LOWER(`corporation_infos`.`tax_rate`) LIKE %a% or LOWER(`corporation_infos`.`member_count`) LIKE %a% or LOWER(`corporation_infos`.`actions`) LIKE %a%)) count_row_table)
(searching something)
```